x/ethbridge/keeper/msg_server.go

```diff
@@ -131,18 +127,25 @@ func (srv msgServer) Burn(goCtx context.Context, msg *types.MsgBurn) (*types.Msg
 		return nil, fmt.Errorf("token metadata not available for %s", msg.DenomHash)
 	}
 
-	prophecyID, err := srv.Keeper.ProcessBurn(ctx, cosmosSender, account.GetSequence(), msg, tokenMetadata)
+	doublePeg := tokenMetadata.NetworkDescriptor != msg.NetworkDescriptor
+	firstDoublePeg := doublePeg && srv.tokenRegistryKeeper.GetFirstLockDoublePeg(ctx, msg.DenomHash, msg.NetworkDescriptor)
+
+	globalSequence := srv.Keeper.GetGlobalSequence(ctx, msg.NetworkDescriptor)
+	prophecyID, err := srv.Keeper.ProcessBurn(ctx, cosmosSender, account.GetSequence(), msg, tokenMetadata, firstDoublePeg)
 
 	if err != nil {
 		logger.Error("bridge keeper failed to process burn.", errorMessageKey, err.Error())
 		return nil, err
 	}
 
-	logger.Info("sifnode emit burn event.", "message", msg)
-	globalSequence := srv.Keeper.GetGlobalSequence(ctx, msg.NetworkDescriptor)
 	srv.Keeper.UpdateGlobalSequence(ctx, msg.NetworkDescriptor, uint64(ctx.BlockHeight()))
 
-	doublePeg := tokenMetadata.NetworkDescriptor != msg.NetworkDescriptor
+	if firstDoublePeg {
+		srv.tokenRegistryKeeper.SetFirstLockDoublePeg(ctx, msg.DenomHash, msg.NetworkDescriptor)
```

Comment:
Should this function be renamed to FirstBurnDoublePeg or just FirstDoublePeg?

Reply:
FirstDoublePeg is fine since double peg only happened when token burned. will update it.